cheroot/test/test_ssl.py

```diff
@@ -1154,3 +1160,82 @@ def test_prepare_socket_emits_deprecation_warning(
     assert sock.fileno() > 0
 
     sock.close()
+
+
+def test_httpconnection_makefile_deprecation(mocker):
+    """
+    Test ``makefile`` argument on ``HTTPConnection`` triggers a warning.
+
+    ``makefile`` is now deprecated.
+    """
+    dummy_server = mocker.create_autospec(HTTPServer, instance=True)
+    dummy_sock = mocker.create_autospec(socket.socket, instance=True)
+
+    # Value for the deprecated 'makefile' parameter
+    dummy_makefile_value = object()
+
+    expected_message = r'makefile.*deprecated'
+
+    # Act & Assert
+    with pytest.deprecated_call(match=expected_message):
+        # Instantiate HTTPConnection, passing the deprecated 'makefile'
+        conn = HTTPConnection(
+            server=dummy_server,
+            sock=dummy_sock,
+            makefile=dummy_makefile_value,  # This triggers the warning
+        )
+
+    # Verify assignment
+    assert conn.server is dummy_server
+    assert conn.socket is dummy_sock
+
+
+@pytest.mark.parametrize(
+    'adapter_type',
+    (
+        'builtin',
+        'pyopenssl',
+    ),
+)
+def test_adapter_makefile_deprecation(
+    mocker,
+    adapter_type,
+    tls_certificate_chain_pem_path,
+    tls_certificate_private_key_pem_path,
+):
+    """Test the adapter's makefile() method emits a deprecation warning."""
+    # Mock the adapter
+    tls_adapter_cls = get_ssl_adapter_class(name=adapter_type)
+    tls_adapter = tls_adapter_cls(
+        tls_certificate_chain_pem_path,
+        tls_certificate_private_key_pem_path,
+    )
+
+    # Create a mock socket with a makefile method
+    dummy_sock = mocker.Mock()
+    mock_file_stream = mocker.Mock(spec=io.FileIO)
+    dummy_sock.makefile.return_value = mock_file_stream
+
+    expected_message = r'makefile.*deprecated'
+
+    # Act & Assert
+    with pytest.deprecated_call(match=expected_message):
+        result = tls_adapter.makefile(dummy_sock, mode='r', bufsize=8192)
+
+    # Assert 1: The correct delegation happened
+    dummy_sock.makefile.assert_called_once_with('r', 8192)
+
+    # Assert 2: The result is the mock file stream
+    assert result is mock_file_stream
+
+
+def test_default_buffer_size_deprecated():
+    """Test accessing ``DEFAULT_BUFFER_SIZE`` raises warning."""
+    with pytest.warns(
+        DeprecationWarning,
+        match='`DEFAULT_BUFFER_SIZE` is deprecated',
+    ):
+        val = builtin_adapter.DEFAULT_BUFFER_SIZE
+
+    # Check that the returned value is correct
+    assert val == io.DEFAULT_BUFFER_SIZE, 'DEFAULT_BUFFER_SIZE value mismatch'
```

Comment:
Does that second `assert` argument add anything of value that `pytest` itself doesn't? It usually is better at this since it's got internal special cases and rewrites for `assert` bytecodes that allow `pytest` extract more detail about the expressions being passed to `assert` and the surrounding variable values + stack frames.
```suggestion
    assert val == io.DEFAULT_BUFFER_SIZE
```